src/index.ts

```diff
@@ -597,9 +589,95 @@ export async function precomputedInit(
     throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
     skipInitialPoll: skipInitialRequest,
   };
-  instance.setSubjectAndPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
 
-  await instance.fetchPrecomputedFlags();
+  const subject: Subject = { subjectKey, subjectAttributes };
+
+  EppoPrecomputedJSClient.instance = new EppoPrecomputedJSClient({
+    precomputedFlagStore: memoryOnlyPrecomputedFlagsStore,
+    requestParameters,
+    subject,
+  });
+
+  EppoPrecomputedJSClient.instance.setAssignmentLogger(config.assignmentLogger);
+  await EppoPrecomputedJSClient.instance.fetchPrecomputedFlags();
+
+  EppoPrecomputedJSClient.initialized = true;
+  return EppoPrecomputedJSClient.instance;
+}
+
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export interface IPrecomputedClientConfigSync {
+  precomputedConfigurationWire: string;
+  assignmentLogger?: IAssignmentLogger;
+  throwOnFailedInitialization?: boolean;
+}
+
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - precomputed client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export function offlinePrecomputedInit(
+  config: IPrecomputedClientConfigSync,
+): EppoPrecomputedClient {
+  const throwOnFailedInitialization = config.throwOnFailedInitialization ?? true;
+
+  const configurationWire: IConfigurationWire = JSON.parse(config.precomputedConfigurationWire);
+  if (!configurationWire.precomputed) {
+    applicationLogger.error('Invalid precomputed configuration wire');
+    return EppoPrecomputedJSClient.instance;
+  }
```

Comment:
This does not comply to `throwOnFailedInitialization`

Reply:
Fixed! An error is thrown if `throwOnFailedInitialization` is true